mobile/native/src/channel_fee.rs

```diff
@@ -128,12 +146,43 @@ impl ChannelFeePaymentSubscriber {
             .expect("Mutex to not be poisoned") = None;
     }
 
-    fn get_funding_transaction(&self) -> Option<EsploraTransaction> {
+    fn get_funding_transaction(&self) -> Option<(ChannelId, EsploraTransaction)> {
         self.open_channel_tx
             .lock()
             .expect("Mutex to not be poisoned")
             .clone()
     }
+
+    async fn wait_for_outbound_capacity(
+        &self,
+        channel_id: ChannelId,
+        funding_tx_fees_msats: u64,
+    ) -> Result<()> {
+        tokio::time::timeout(Duration::from_secs(5), async {
+            loop {
+                let channel_details = self
+                    .channel_manager
+                    .get_channel_details(&channel_id)
+                    .expect("Channel details to exist");
+
+                if channel_details.outbound_capacity_msat >= funding_tx_fees_msats {
+                    tracing::debug!(channel_details.outbound_capacity_msat, channel_id=hex::encode(channel_id),
+                        "Channel has enough outbound capacity");
+                    break;
+                } else {
+                    tracing::debug!(channel_id = hex::encode(channel_id), outbound_capacity_msats = channel_details.outbound_capacity_msat, funding_tx_fees_msats,
+                        "Channel does not have enough outbound capacity to pay jit channel opening fees yet. Waiting.");
+                    tokio::time::sleep(Duration::from_millis(200)).await
+                }
+            }
+        })
+        .await
+        .map_err(|e| anyhow!("{e:#}"))
+        .context(format!(
```

Comment:
🔧 Should use `with_context` instead.

```suggestion
        .with_context(|| format!(
```

Reply:
well.. happy to change it, but do you think printing the text is such a huge performance cost, that we have to do it lazily? or do you have other reasons for suggesting that?

Reply:
I'm sure it's negligible, but I think it's more correct to use `with_context` when you have to build the error so that it is evaluated lazily.

So obviously not a big deal at all, but I wouldn't know why not to do it.

Reply:
@holzeis just pointing out that I didn't mean to say to _always_ use `with_context`. It depends on the... context 😛

Reply:
`format!` itself also incurs an allocation since it makes a String